build/pause/Makefile

```diff
@@ -12,20 +12,35 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-.PHONY:	build push
+CFLAGS = -Os -Wall -static
 
-TAG = 2.0
+ARCH ?= amd64
 
-ARCH = amd64
-IMAGE = pause
-# ARCH = arm
-# IMAGE = pause-arm
+ifeq ($(ARCH),amd64)
+	TRIPLE ?= x86_64-linux-gnu
+endif
 
-build:
-	./prepare.sh ${ARCH}
-	docker build -t gcr.io/google_containers/$(IMAGE):$(TAG) .
+ifeq ($(ARCH),arm)
+	TRIPLE ?= arm-linux-gnueabi
+endif
 
-push:	build
-	gcloud docker --server=gcr.io push gcr.io/google_containers/$(IMAGE):$(TAG)
+IMAGE = gcr.io/google_containers/pause-$(ARCH)
+TAG = 3.0
 
-all:	push
+pause: pause.c
+	docker run -v $(CURDIR):/build gcr.io/google_containers/kube-cross:v1.4.2-1 \
+		/bin/bash -c "\
+			cd /build && \
+			$(TRIPLE)-gcc $(CFLAGS) -o pause $< && \
+			$(TRIPLE)-strip pause"
+
+build: pause
+	docker build -t $(IMAGE):$(TAG) .
+
+push: build
```

Comment:
we should probably start setting a better example.  This is a bit clunky.  We should do something like:

```
container .container: $(BIN)
  X=$$(docker build ...) &&  echo $$X > .container

push: .container
  docker push ...
```

don't forget to clean up .container